### PR TITLE
BaSP-TG-145:  Create Header Component

### PR DESCRIPTION
**Pull request by CristianBucci**
- Create Header component
- Add in index routes
- Add sign in/login button
- When logged, change to the logout button

QA:

To test this PR you should see the header in all the screen you navigate, and once logged the sign-up and the login button should not be displayed, but the login button yes! 

Note: The TrackGenix logo doesn't work yet, this will be solved in incoming PR!

### CODE REVIEW

**Review by renzopimentel — APPROVED** [review]
Looks great

File: src/Components/Header/index.js [hunk 1]
```diff
@@ -1,62 +1,61 @@
...
+    <>
+      <ModalConfirm
+        show={showConfirmModal}
+        modalTitle={modalContent.title}
+        modalContent={modalContent.content}
+        onConfirm={onConfirm}
+        onCancel={onCancel}
+      />
+      <header className={styles.header}>
+        <Link style={{ textDecoration: 'none' }} to="/home">
```
Comment (Ansitox) [inline src/Components/Header/index.js:37]: Move the style to de css file

File: src/Components/Header/index.js [hunk 2]
```diff
@@ -1,62 +1,61 @@
...
+
+  const dispatch = useDispatch();
+
+  const onConfirm = () => {
+    dispatch(logout());
+    dispatch(confirmModalClose());
+  };
+
+  const onSubmit = () => {
+    const content = `Are you sure you want to logout?`;
```
Comment (Ansitox) [inline src/Components/Header/index.js:19]: Change the backticks

File: src/Components/Header/index.js [hunk 3]
```diff
@@ -1,62 +1,61 @@
...
-      </nav>
-    </header>
+        <div className={styles.buttonContainer}>
+          {isAuthenticated ? (
+            <button className={styles.headerButton} onClick={onSubmit}>
+              LOG OUT
+            </button>
+          ) : (
+            <>
+              <Link style={{ textDecoration: 'none' }} to="/auth/login">
```
Comment (Ansitox) [inline src/Components/Header/index.js:47]: Same as above

**Review by Ansitox — APPROVED** [review]
Look great

**Review by julianpiaggio — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK

We wanted to tell you that we consider that your pr complies
the requirements to be approved, and that it can be merged into the master branch 👍

**Pull request merged**